### PR TITLE
Update docker

### LINKED ISSUE #110856: Docker for Mac brew cask has a SHA256 mismatch
### 

- [X] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [X] I have retried my command with `--force`.
- [X] I ran `brew update-reset && brew update` and retried my command.
- [X] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [X] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [ ] I made doubly sure this is not a [checksum does not match](https://github.com/Homebrew/homebrew-cask/blob/master/doc/reporting_bugs/checksum_does_not_match_error.md) error.

### Description of issue

The expected and actual [SHA256's for Docker for Mac cask](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/docker.rb#L5) for `amd64` platforms are not matching up
```
Error: SHA256 mismatch
Expected: 08c91be8f784c41b677b28f6eea2a7ec001e0b812bb0ee978d1c0b60ecea77bb
  Actual: 6b1d73f4b9d3be08798a959255726fc8bf3e9ee120c9e3ab8ac8c38b42dc8675
```

### Command that failed

brew install --cask docker

### Output of command with `--verbose --debug`

<details><summary>Click to expand</summary><pre><code>

$ brew install --cask docker --verbose
==> Downloading https://desktop.docker.com/mac/stable/amd64/67351/Docker.dmg
Already downloaded: /Users/josephschmitt/Library/Caches/Homebrew/downloads/103511eaed33ac464077411746dfcd9af769d3bbf75dba85e6cda8c3b4760366--Docker.dmg
==> Verifying checksum for cask 'docker'
Error: SHA256 mismatch
Expected: 08c91be8f784c41b677b28f6eea2a7ec001e0b812bb0ee978d1c0b60ecea77bb
  Actual: 6b1d73f4b9d3be08798a959255726fc8bf3e9ee120c9e3ab8ac8c38b42dc8675
    File: /Users/josephschmitt/Library/Caches/Homebrew/downloads/103511eaed33ac464077411746dfcd9af769d3bbf75dba85e6cda8c3b4760366--Docker.dmg
To retry an incomplete download, remove the file above.

</code></pre></details>

### Output of `brew doctor --verbose`

<details><summary>Click to expand</summary><pre><code>

$ brew doctor --verbose
==> Cask Environment Variables:
BUNDLE_PATH
CHRUBY_VERSION
GEM_HOME
GEM_PATH
HOMEBREW_CASK_OPTS
LC_ALL
PATH
RBENV_VERSION
RUBYLIB
RUBYOPT
RUBYPATH
SHELL
==> $LOAD_PATHS
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/warning-1.2.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/tapioca-0.4.25/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/spoom-1.1.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/thor-1.1.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/sorbet-runtime-stub-0.2.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/simplecov-cobertura-1.4.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/simplecov-0.21.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/simplecov_json_formatter-0.1.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/simplecov-html-0.12.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ruby-macho-2.5.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-sorbet-0.6.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-rspec-2.4.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-rails-2.11.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-performance-1.11.5/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-1.20.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/unicode-display_width-2.0.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ruby-progressbar-1.11.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-ast-1.11.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec_junit_formatter-0.4.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-wait-0.0.9/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-sorbet-1.8.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/sorbet-0.5.9088/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/sorbet-static-0.5.9088-universal-darwin-14/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-retry-0.6.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-its-1.3.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-github-2.3.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-3.10.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-mocks-3.10.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-expectations-3.10.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-core-3.10.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-support-3.10.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ronn-0.7.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rexml-3.2.5/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/regexp_parser-2.1.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rdiscount-2.2.0.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/extensions/x86_64-darwin-14/2.6.0-static/rdiscount-2.2.0.2
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rack-2.2.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/pry-0.14.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/plist-3.6.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/patchelf-1.3.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/parlour-6.0.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/sorbet-runtime-0.5.9088/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rainbow-3.0.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/parser-3.0.2.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/parallel_tests-3.7.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/parallel-1.20.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mustache-1.1.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/method_source-1.0.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mechanize-2.8.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/webrobots-0.1.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/webrick-1.7.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubyntlm-0.6.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/nokogiri-1.11.7-x86_64-darwin/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/racc-1.5.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/extensions/x86_64-darwin-14/2.6.0-static/racc-1.5.2
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mini_portile2-2.5.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/net-http-persistent-4.0.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/net-http-digest_auth-1.4.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mime-types-3.3.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mime-types-data-3.2021.0704/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/http-cookie-1.0.4/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/hpricot-0.8.6/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/extensions/x86_64-darwin-14/2.6.0-static/hpricot-0.8.6
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/elftools-1.1.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/domain_name-0.5.20190701/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/unf-0.1.4/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/unf_ext-0.0.7.7/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/extensions/x86_64-darwin-14/2.6.0-static/unf_ext-0.0.7.7
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/docile-1.4.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/diff-lcs-1.4.4/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/did_you_mean-1.5.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/connection_pool-2.2.5/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/commander-4.6.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/highline-2.0.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/colorize-0.8.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/coderay-1.1.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/byebug-11.1.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/extensions/x86_64-darwin-14/2.6.0-static/byebug-11.1.3
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/bootsnap-1.8.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/extensions/x86_64-darwin-14/2.6.0-static/bootsnap-1.8.1
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/msgpack-1.4.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/extensions/x86_64-darwin-14/2.6.0-static/msgpack-1.4.2
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/bindata-2.4.10/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ast-2.4.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/addressable-2.8.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/public_suffix-4.0.6/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/activesupport-6.1.4.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/zeitwerk-2.4.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/tzinfo-2.0.4/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/minitest-5.14.4/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/i18n-1.8.10/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/concurrent-ruby-1.1.9/lib/concurrent-ruby
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/gems/2.6.0/gems/did_you_mean-1.3.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/site_ruby/2.6.0
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/site_ruby/2.6.0/x86_64-darwin14
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/site_ruby
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/vendor_ruby/2.6.0
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/vendor_ruby/2.6.0/x86_64-darwin14
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/vendor_ruby
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/x86_64-darwin14
/usr/local/Homebrew/Library/Homebrew
==> Homebrew Version
3.2.10-22-gd429780
==> macOS
11.5.2
==> SIP
Enabled
==> Homebrew Cask Staging Location
/usr/local/Caskroom
==> Homebrew Cask Taps:
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask (3921 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-versions (195 casks)
/usr/local/Homebrew/Library/Taps/adoptopenjdk/homebrew-openjdk (47 casks)
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed with `brew diy`.
You should find replacements for the following formulae:
  mongodb@3.4

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  lefthook
  proj@7

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  coreutils
  docker-compose
  python@3.9

Warning: Some installed formulae are not readable:
  mongodb@3.4: unknown or unsupported macOS version: :mountain_lion

Warning: You have uncommitted modifications to Homebrew/brew.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /usr/local/Homebrew && git stash && git clean -d -f

Uncommitted files:
  ?? Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/codecov-0.4.3/lib/codecov.rb
  ?? Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/codecov-0.4.3/lib/codecov/formatter.rb
  ?? Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/codecov-0.4.3/lib/codecov/uploader.rb
  ?? Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/codecov-0.4.3/lib/codecov/version.rb

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'fish_add_path /usr/local/sbin' >> ~/.config/fish/config.fish

Warning: Broken symlinks were found. Remove them with `brew cleanup`:
  /usr/local/bin/com.docker.cli
  /usr/local/bin/docker-compose-v1
  /usr/local/bin/hub-tool
  /usr/local/lib/docker/cli-plugins

Warning: Some installed formulae are missing dependencies.
You should `brew install` the missing dependencies:
  brew install node

Run `brew missing` for more details

</code></pre></details>

### Output of `brew tap`

<details><summary>Click to expand</summary><pre><code>

$ brew tap
adoptopenjdk/openjdk
arkweid/lefthook
github/gh
homebrew/bundle
homebrew/cask
homebrew/cask-versions
homebrew/core
homebrew/services
urbancompass/versions

</code></pre></details>

### PR DESCRIPTION
**Pull request by josephschmitt**
Created with `brew bump-cask-pr`.

Fixes #110856

### CODE REVIEW

File: Casks/docker.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
   version "3.6.0,67351"
 
   if Hardware::CPU.intel?
-    sha256 "08c91be8f784c41b677b28f6eea2a7ec001e0b812bb0ee978d1c0b60ecea77bb"
+    sha256 "409ca31a47ba1d1620825f09e02f6d5ce203b2907ab91047041a0e9b4b719593"
```
Comment (jonathanbettger) [inline Casks/docker.rb:5]: This looks to be the SHA of the 4.0.0 dmg, not the updated 3.6.0 SHA.

Comment (josephschmitt) [reply Casks/docker.rb:5]: Hmm, that's what I got from their downloads page. any idea where I can try and download other versions?

Comment (WillAbides) [reply Casks/docker.rb:5]: `brew bump-cask-pr -n --version 3.6.0,67351 --force docker` gave me this output:

```
...
==> replace /version\s+["']3\.6\.0,67351["']/m with "version \"3.6.0,67351\""
==> replace "61d97003dfcc0cbe49efa4439f1d0bfb899297a79693ebb9916c3e3420309f47" with "8d03e6111d377d1bd366cb7bc1bf90b072f0f0eab9541b125d7ceb9c0ffe00f5"
==> replace /sha256\s+["']08c91be8f784c41b677b28f6eea2a7ec001e0b812bb0ee978d1c0b60ecea77bb["']/m with "sha256 \"6b1d73f4b9d3be08798a959255726fc8bf3e9ee120c9e3ab8ac8c38b42dc8675
==> brew audit --cask docker.rb
==> brew style --fix docker.rb
==> try to fork repository with GitHub API
==> git add /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/docker.rb
==> git checkout --no-track -b bump-docker-3.6.0-67351 origin/master
==> git commit --no-edit --verbose --message='Update docker from 3.6.0,67351 to 3.6.0,67351' -- /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/docker.rb
==> git push --set-upstream FORK_URL bump-docker-3.6.0-67351:bump-docker-3.6.0-67351
==> git checkout --quiet -
==> create pull request with GitHub API (base branch: master)
```

Note the shas on lines 2 and 3.

**Comment by dalisoft** [comment]
Hey and thank you for PR.

There are **Docker v4.0.0** released, maybe would be better to update to **v4** instead of just updating hashesh?

File: Casks/docker.rb (hunk 1 above)
Comment (josephschmitt) [reply Casks/docker.rb:5]: Found it, I'll update now https://docs.docker.com/desktop/mac/release-notes/

Comment (WillAbides) [reply Casks/docker.rb:5]: @josephschmitt To answer your question directly, the version 3.6.0,67351 downloads are at 

https://desktop.docker.com/mac/stable/amd64/67351/Docker.dmg 
and
https://desktop.docker.com/mac/stable/arm64/67351/Docker.dmg

**Comment by WillAbides** [comment]
It's curious that docker has apparently updated the Docker.dmg without iterating the version for 3.6.0.

File: Casks/docker.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@
       strategy :sparkle
     end
   else
-    sha256 "61d97003dfcc0cbe49efa4439f1d0bfb899297a79693ebb9916c3e3420309f47"
+    sha256 "1b3b5facbe8e05f582382e8ab24073736720e64c6a66c3a594759f98a854b59d"
 
     url "https://desktop.docker.com/mac/stable/arm64/#{version.after_comma}/Docker.dmg"
```
Comment (dalisoft) [inline Casks/docker.rb:16]:
```suggestion
    url "https://desktop.docker.com/mac/stable/arm64/Docker.dmg"
```

File: Casks/docker.rb [hunk 3]
```diff
@@ -11,7 +11,7 @@
       strategy :sparkle
     end
   else
-    sha256 "61d97003dfcc0cbe49efa4439f1d0bfb899297a79693ebb9916c3e3420309f47"
+    sha256 "1b3b5facbe8e05f582382e8ab24073736720e64c6a66c3a594759f98a854b59d"
```
Comment (dalisoft) [inline Casks/docker.rb]: This hashes for **v4** only as i understood from file

**Comment by josephschmitt** [comment]
> Hey and thank you for PR.
> 
> There are **Docker v4.0.0** released, maybe would be better to update to **v4** instead of just updating hashesh?

Hmm, to me it seems safer to address this specific mis-match first, before committing to pointing to a breaking change version, no?

File: Casks/docker.rb (hunk 1 above)
Comment (josephschmitt) [reply Casks/docker.rb]: Yup, updated my PR, thanks

**Comment by dalisoft** [comment]
> > Hey and thank you for PR.
> > There are **Docker v4.0.0** released, maybe would be better to update to **v4** instead of just updating hashesh?
> 
> Hmm, to me it seems safer to address this specific mis-match first, before committing to pointing to a breaking change version, no?

Yes, you right. Sorry. I did not think about that

**Comment by josephschmitt** [comment]
> It's curious that docker has apparently updated the Docker.dmg without iterating the version for 3.6.0.

I'm also curious about this. I wonder if they just updated the packager itself (as opposed to the actual app) and that was enough to trigger a new sha.

**Review by WillAbides — APPROVED** [review]
Thanks for taking the time to update this

**Comment by josephschmitt** [comment]
> Thanks for taking the time to update this

Happy to help! It was also blocking a bunch of our developers who were on-boarding today, so I was highly motivated to fix :-)

**Comment by josephschmitt** [comment]
@WillAbides anyone else I need to ping for review? This is kind of a major blocker for us right now so would be great to get merged ASAP.

**Comment by WillAbides** [comment]
@josephschmitt I don't know who needs to approve this.

If people are stuck now you can have them do this:

```
cd "$(brew --repo)/Library/Taps/homebrew/homebrew-cask"
git remote add josephschmitt "https://github.com/josephschmitt/homebrew-cask.git"
git fetch josephschmitt
git checkout bump-docker
brew install --cask docker
```

**Comment by jamiegs** [comment]
Seems like this might have been to push out their new business subscription for Docker Desktop. 😞

**Comment by nicolasbarra** [comment]
Should there be another PR to reflect the change to 4.0? See [here](https://docs.docker.com/desktop/mac/release-notes/).

**Comment by bevanjkay** [comment]
Merging this to get the cask working, we will need to update to 4.0 – however the current livecheck is not returning version `4.0` so that would need to be updated also

**Pull request merged**